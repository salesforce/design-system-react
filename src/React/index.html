<!doctype html public "library">
<meta charset=utf-8/>
<head>
	<title>Selectlist</title>
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
</head>
<body>
<script src="/dist/react/build.js"></script>
</body>
</html>