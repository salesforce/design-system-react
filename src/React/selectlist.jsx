//test

var React = require('react');
import {SelectlistCore} from "../SelectlistCore";
var DropdownButton = require('react-bootstrap/lib/DropdownButton');
var MenuItem = require('react-bootstrap/lib/MenuItem');

var DATA = {
	selected: 'None selected',
	id: 'selectlist1',
	items: [
		{ id: 0, name: 'tacos', type: 'mexican' },
		{ id: 1, name: 'burrito', type: 'mexican' },
		{ id: 2, name: 'tostada', type: 'mexican' },
		{ id: 3, name: 'hush puppies', type: 'southern' }
	]
};

var menuItemSelected = (eventKey, href, target)=> {
	DATA.selected = DATA.items[eventKey].name;
};
//Selectlist.handleMenuItemClicked
var menuItems = DATA.items.map( (menuItem) => {
	return <MenuItem eventKey={menuItem.id}>{menuItem.name}</MenuItem>;
});

var Selectlist = React.createClass({
	getInitialState: function() {
		return {
			selectedKey: undefined
		};
	},

	handleMenuItemClicked: function(eventKey, href, target) {
		this.setState({
			selectedKey: eventKey
		});
	},

	menuItems: function() {
		return DATA.items.map( (menuItem) => {
			return <MenuItem eventKey={menuItem.id} onSelect={this.handleMenuItemClicked}>{menuItem.name}</MenuItem>;
		});
	},

	selected: function() {
		return typeof(this.state.selectedKey) === 'undefined' ? 'None Selected' : DATA.items[this.state.selectedKey].name;
	},

	render () {
		return (
			<div>
				<h1>SelectList</h1>
				<ul>{this.menuItems}</ul>
				<DropdownButton title={this.selected()} key={DATA.id}>{this.menuItems()}</DropdownButton>
			</div>
		);
	}
});

React.render(<Selectlist/>, document.body, () => {});