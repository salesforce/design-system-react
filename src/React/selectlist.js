// SELECTLIST CONTROL

// core
import {Landmark} from '../Landmark';
import {SelectlistCore} from '../SelectlistCore';

// framework specific
import React from 'react';
import DropdownButton from 'react-bootstrap/lib/DropdownButton';
import MenuItem from 'react-bootstrap/lib/MenuItem';

export var Selectlist = React.createClass(Object.assign({}, SelectlistCore, {	
	propTypes: {
		disabled: React.PropTypes.bool,
		selection: React.PropTypes.number,
		collection: React.PropTypes.array
	},
	
	getInitialState () {
		return this.__getInitialState();
	},
		
	menuItems () {
		return this.props.collection.map((menuItem) => {
			return <MenuItem eventKey={menuItem.id} onSelect={this.handleMenuItemClicked}>
					{menuItem.name}
				</MenuItem>;
		});
	},
	
	componentWillReceiveProps(nextProps) {
		this.__initializeOptions(nextProps);
	},
	
	render () {
		var selection = this.getSelection();
		
		return (
			<DropdownButton className={this.cssClasses.CONTROL} disabled={this.state.disabled} title={selection ? selection.name : 'None selected'} key={this.props.id}>{this.menuItems()}</DropdownButton>
		);
	},
	
	componentWillMount () {
		this.elements = {
			wrapper: {
				toggleClass: function (cssClass, state) {
					Landmark.log(cssClass, state);
				}
			}
		}
		
		this.__constructor(this.props);
	},
	
	handleMenuItemClicked (eventKey, href, target) {
		this.setSelectionByKey('id', eventKey);
	}
}));