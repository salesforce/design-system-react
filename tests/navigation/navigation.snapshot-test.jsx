import React from 'react';
import renderer from 'react-test-renderer';
import { renderMarkup } from '../snapshot-helpers';

import SnapshotDefault from '../../examples/navigation/snapshot-default';

import globalSettings from '../../components/settings';

test('Navigation Default DOM Snapshot', () => {
	const domTree = renderer.create(
		<SnapshotDefault />,
	).toJSON();
	expect(domTree).toMatchSnapshot();
});

test('Navigation Default HTML Snapshot', () => {
	expect(renderMarkup(SnapshotDefault)).toMatchSnapshot();
});

const customProps = {
	className: 'CUSTOM-CLASSNAME',
	id: 'CUSTOM-ID',
	variant: 'shade',
	selectedId: 'all_reports'
};

test(`Navigation
		className,
		id,
		variant
	DOM Snapshot`, () => {
	const domTree = renderer.create(
		<SnapshotDefault
			{...customProps}
		/>,
	).toJSON();
	expect(domTree).toMatchSnapshot();
});
