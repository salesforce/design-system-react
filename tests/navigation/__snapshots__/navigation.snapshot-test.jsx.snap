// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation
		className,
		id,
		variant
	DOM Snapshot 1`] = `
<div
  className="slds-grid slds-grid--vertical slds-navigation-list--vertical slds-navigation-list--vertical-inverse CUSTOM-CLASSNAME"
  id="CUSTOM-ID"
>
  <h2
    className="slds-text-title--caps slds-p-around--medium"
    id="CUSTOM-ID-reports"
  >
    Reports
  </h2>
  <ul>
    <li
      className=""
    >
      <a
        aria-describedby="CUSTOM-ID-reports"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="recent_reports"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        Recent
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-describedby="CUSTOM-ID-reports"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="my_reports"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        Created by Me
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-describedby="CUSTOM-ID-reports"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="private_reports"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        Private Reports
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-describedby="CUSTOM-ID-reports"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="public_reports"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        Public Reports
      </a>
    </li>
    <li
      className="slds-is-active"
    >
      <a
        aria-describedby="CUSTOM-ID-reports"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="all_reports"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        All Reports
      </a>
    </li>
  </ul>
  <h2
    className="slds-text-title--caps slds-p-around--medium"
    id="CUSTOM-ID-folders"
  >
    Folders
  </h2>
  <ul>
    <li
      className=""
    >
      <a
        aria-describedby="CUSTOM-ID-folders"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="my_folders"
        href="http://www.google.com"
        onClick={[Function]}
      >
        Created by Me
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-describedby="CUSTOM-ID-folders"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="shared_folders"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        Shared with Me
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-describedby="CUSTOM-ID-folders"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="all_folders"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        All Folders
      </a>
    </li>
  </ul>
</div>
`;

exports[`Navigation Default DOM Snapshot 1`] = `
<div
  className="slds-grid slds-grid--vertical slds-navigation-list--vertical"
  id="sample-navigation"
>
  <h2
    className="slds-text-title--caps slds-p-around--medium"
    id="sample-navigation-reports"
  >
    Reports
  </h2>
  <ul>
    <li
      className="slds-is-active"
    >
      <a
        aria-describedby="sample-navigation-reports"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="recent_reports"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        Recent
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-describedby="sample-navigation-reports"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="my_reports"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        Created by Me
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-describedby="sample-navigation-reports"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="private_reports"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        Private Reports
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-describedby="sample-navigation-reports"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="public_reports"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        Public Reports
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-describedby="sample-navigation-reports"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="all_reports"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        All Reports
      </a>
    </li>
  </ul>
  <h2
    className="slds-text-title--caps slds-p-around--medium"
    id="sample-navigation-folders"
  >
    Folders
  </h2>
  <ul>
    <li
      className=""
    >
      <a
        aria-describedby="sample-navigation-folders"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="my_folders"
        href="http://www.google.com"
        onClick={[Function]}
      >
        Created by Me
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-describedby="sample-navigation-folders"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="shared_folders"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        Shared with Me
      </a>
    </li>
    <li
      className=""
    >
      <a
        aria-describedby="sample-navigation-folders"
        className="slds-navigation-list--vertical__action slds-text-link--reset"
        data-id="all_folders"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        All Folders
      </a>
    </li>
  </ul>
</div>
`;

exports[`Navigation Default HTML Snapshot 1`] = `
"<div id=\\"sample-navigation\\" class=\\"slds-grid slds-grid--vertical slds-navigation-list--vertical\\">
  <h2 id=\\"sample-navigation-reports\\" class=\\"slds-text-title--caps slds-p-around--medium\\">Reports</h2>
  <ul>
    <li class=\\"slds-is-active\\"><a data-id=\\"recent_reports\\" href=\\"javascript:void(0);\\" class=\\"slds-navigation-list--vertical__action slds-text-link--reset\\" aria-describedby=\\"sample-navigation-reports\\">Recent</a></li>
    <li class=\\"\\"><a data-id=\\"my_reports\\" href=\\"javascript:void(0);\\" class=\\"slds-navigation-list--vertical__action slds-text-link--reset\\" aria-describedby=\\"sample-navigation-reports\\">Created by Me</a></li>
    <li class=\\"\\"><a data-id=\\"private_reports\\" href=\\"javascript:void(0);\\" class=\\"slds-navigation-list--vertical__action slds-text-link--reset\\" aria-describedby=\\"sample-navigation-reports\\">Private Reports</a></li>
    <li class=\\"\\"><a data-id=\\"public_reports\\" href=\\"javascript:void(0);\\" class=\\"slds-navigation-list--vertical__action slds-text-link--reset\\" aria-describedby=\\"sample-navigation-reports\\">Public Reports</a></li>
    <li class=\\"\\"><a data-id=\\"all_reports\\" href=\\"javascript:void(0);\\" class=\\"slds-navigation-list--vertical__action slds-text-link--reset\\" aria-describedby=\\"sample-navigation-reports\\">All Reports</a></li>
  </ul>
  <h2 id=\\"sample-navigation-folders\\" class=\\"slds-text-title--caps slds-p-around--medium\\">Folders</h2>
  <ul>
    <li class=\\"\\"><a data-id=\\"my_folders\\" href=\\"http://www.google.com\\" class=\\"slds-navigation-list--vertical__action slds-text-link--reset\\" aria-describedby=\\"sample-navigation-folders\\">Created by Me</a></li>
    <li class=\\"\\"><a data-id=\\"shared_folders\\" href=\\"javascript:void(0);\\" class=\\"slds-navigation-list--vertical__action slds-text-link--reset\\" aria-describedby=\\"sample-navigation-folders\\">Shared with Me</a></li>
    <li class=\\"\\"><a data-id=\\"all_folders\\" href=\\"javascript:void(0);\\" class=\\"slds-navigation-list--vertical__action slds-text-link--reset\\" aria-describedby=\\"sample-navigation-folders\\">All Folders</a></li>
  </ul>
</div>"
`;
